### PR TITLE
feat(api): implement session management with clear_session and clear_all_sessions methods

### PR DESCRIPTION
**Pull request by RaHehl**
### Description of change

Since there are occasional cases where manually deleting the session data is necessary, this should serve as the basis for automatically removing these data when the integration is removed, so that no one has to delete files manually anymore.

### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] This pull request follows the [contributing guidelines](https://github.com/uilibs/uiprotect/blob/main/CONTRIBUTING.md).
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/), such as "fix(api): prevent racing of requests".

> - If pre-commit.ci is failing, try `pre-commit run -a` for further information.
> - If CI / test is failing, try `poetry run pytest` for further information.

## Summary by CodeRabbit

* **New Features**
  * Added actions to clear the current session or remove all stored sessions; these reset authentication state and stored tokens.

* **Bug Fixes**
  * Clear actions are no-ops when session storage is disabled and handle missing/invalid session data or concurrent removal gracefully; CSRF/session edge cases handled without raising errors.

* **Tests**
  * Expanded coverage for session loading, invalid/partial tokens, CSRF presence and rotation, reauthentication flows, and robust session-storage edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/uiprotect/api.py [hunk 1]
```diff
@@ -863,6 +863,55 @@ def is_authenticated(self) -> bool:
...
+
+        config: dict[str, Any] = {}
+        session_hash = get_user_hash(str(self._url), self._username)
+        try:
+            async with aiofiles.open(self.config_file, "rb") as f:
+                config_data = await f.read()
+                if config_data:
+                    try:
+                        config = orjson.loads(config_data)
+                    except Exception:
```
Comment (bdraco) [inline src/uiprotect/api.py]: should avoid broad exception catch

File: src/uiprotect/api.py [hunk 2]
```diff
@@ -863,6 +863,55 @@ def is_authenticated(self) -> bool:
...
+
+        try:
+            if self.config_file.exists():
+                await aos.remove(self.config_file)
+                _LOGGER.debug("Cleared all sessions from config file")
+
+                # Clear authentication state only after successful deletion
+                self._is_authenticated = False
+                self._last_token_cookie = None
+                self._last_token_cookie_decode = None
```
Comment (bdraco) [inline src/uiprotect/api.py]: could move to else:

File: src/uiprotect/api.py [hunk 3]
```diff
@@ -863,6 +863,55 @@ def is_authenticated(self) -> bool:
...
+            self._last_token_cookie = None
+            self._last_token_cookie_decode = None
+
+    async def clear_all_sessions(self) -> None:
+        """Clears all stored sessions from the config file."""
+        if not self.store_sessions:
+            return
+
+        try:
+            if self.config_file.exists():
```
Comment (bdraco) [inline src/uiprotect/api.py]: technically a TOCTOU race here

Comment (bdraco) [reply src/uiprotect/api.py]: 65e44e6525e4dbf3498f31649e9dbe479fb1c743

File: src/uiprotect/api.py (hunk 2 above)
Comment (bdraco) [reply src/uiprotect/api.py]: 65e44e6525e4dbf3498f31649e9dbe479fb1c743

File: src/uiprotect/api.py (hunk 1 above)
Comment (bdraco) [reply src/uiprotect/api.py]: 65e44e6525e4dbf3498f31649e9dbe479fb1c743
